===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(p11_index_take:obj)) at 25 (line 518, column 2) }
===== Actual patch in diff format =====
@@ -515,7 +515,11 @@ p11_index_take (p11_index *index,
 		return rv;
 	}
 
-	return_val_if_fail (obj->attrs != NULL, CKR_GENERAL_ERROR);
+	if (obj->attrs != NULL) { } else {
+		p11_debug_precond ("p11-kit: '%s' not true at %s\n", "obj->attrs != NULL", __func__);
+		free(obj);
+		return CKR_GENERAL_ERROR;
+	}
 
 	if (!p11_dict_set (index->objects, &obj->handle, obj))
 		return_val_if_reached (CKR_HOST_MEMORY);
